### ISSUE TITLE
Modify loss function or mask attention in this code

### ISSUE THREAD

**Issue by madhurapande19** [OpenNMT/OpenNMT-tf#970]
I want to use this code to experiment with loss function and attention mechanism. Is it possible to build and run this from its source code (instead of command line) where I can make changes and test? If yes, can you please tell me the steps?

Thanks

**Comment by guillaumekln** [comment]
You can can clone the repository and install the library from the source directory:

```bash
pip install -e .
```

Here are some pointers in the code:

* [`MultiHeadAttention`](https://github.com/OpenNMT/OpenNMT-tf/blob/v2.28.0/opennmt/layers/transformer.py#L205-L387) module
* [`cross_entropy_sequence_loss`](https://github.com/OpenNMT/OpenNMT-tf/blob/v2.28.0/opennmt/utils/losses.py#L28-L96) function (called by the model [`compute_loss`](https://github.com/OpenNMT/OpenNMT-tf/blob/v2.28.0/opennmt/models/sequence_to_sequence.py#L416-L466) method)
